### PR TITLE
[ols] Add id to returned edges

### PR DESCRIPTION
**Pull request by lcpopa**

### CODE REVIEW

**Review by lpalashevski — APPROVED** [review]

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/graph/LineageGraphQueryHelper.java [hunk 1]
```diff
@@ -101,12 +101,22 @@ public Set<LineageEdge> getLineageEdges(Graph subGraph) {
...
         while (originalEdges.hasNext()) {
             Edge next = originalEdges.next();
-            LineageEdge newLineageEdge = new LineageEdge(next.label(), getNodeID(next.outVertex()), getNodeID(next.inVertex()));
+            LineageEdge newLineageEdge = new LineageEdge(getEdgeID(next), next.label(), getNodeID(next.outVertex()), getNodeID(next.inVertex()));
             lineageEdges.add(newLineageEdge);
         }
         return lineageEdges;
     }
 
+    private String getEdgeID(Edge edge) {
```
Comment (davidradl) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/graph/LineageGraphQueryHelper.java:110]: I suggest some comments around this to explain the thinking behind this.

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-api/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/model/LineageEdge.java [hunk 2]
```diff
@@ -27,16 +27,21 @@ public class LineageEdge implements Serializable {
 
     public LineageEdge(){}
 
+    protected Object id;
```
Comment (davidradl) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-api/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/model/LineageEdge.java:30]: why is this an Object - is it not always a String?

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/graph/LineageGraphQueryHelper.java (hunk 1 above)
Comment (lcpopa) [reply open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/graph/LineageGraphQueryHelper.java:110]: Until now, edges were not sent to the client with their id. This method computes an edge id in a way similar to computing a node id

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-api/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/model/LineageEdge.java (hunk 2 above)
Comment (lcpopa) [reply open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-api/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/model/LineageEdge.java:30]: Consistency is the reason for having id as an Object. Check LineageVertex

**Pull request merged**